fitter/fit_data.py

```diff
@@ -0,0 +1,287 @@
+        Number of coefficients.
+
+    Returns
+    -------
+    coeffs : `list` [`float`]
+        The fit coefficients.
+    covariance : `numpy.ndarray`
+        Estimated covariance of coeffs. The diagonals provide the variance
+        of the parameter estimate. See the documentation for
+        ``scipy.optimize.curve_fit`` for more information.
+    """
+    if ncoeffs < 1:
+        raise ValueError("Must be at least 1 coeff")
+    cause_data = data["cause"]
+    effect_data = data[axis]
+    coeffs, covariance = scipy.optimize.curve_fit(
+        model, cause_data, effect_data, p0=[0] * ncoeffs
+    )
+    return coeffs, covariance
+
+
+def fit_all_axes(data, cause, model, ncoeffs_arr, print_covariance=False, graph_path=None):
+    """Fit and plot all AXES_TO_FIT using varying numbers of coeffs.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Data as a structured array with fields: elevation (deg)
+        plus all fields in COLUMN_NAMES
+    cause : `str`
+        Name of cause being compensated for, e.g. "elevation".
+        Note that `read_data` converts zd to elevation,
+        so you should do the same for this argument.
+    model : callable
+        Model to fit. It must take the following arguments:
+
+        * ang: a numpy.ndarray of angles, in deg
+        * *coeffs: a sequence of coefficients
+    ncoeffs_arr : `list` [`int`]
+        Sequence of numbers of coefficients to try.
+    print_covariance : `bool`, optional
+        Print the covariance matrix from the fit?
+    graph_path : `str` or `None`, optional
+        Path to which to save the graph.
+        If `None` do not save the graph and pause to show it.
+    """
+    cause_data = data["cause"]
+    cause_units = UNITS_DICT[cause]
+    dense_cause_data = np.linspace(start=cause_data[0], stop=cause_data[-1])
+
+    nfits = len(AXES_TO_FIT)
+    fig, plots = plt.subplots(2, nfits, sharex=True, figsize=(20, 10))
+    plot_dict = {axis: (plots[0, i], plots[1, i]) for i, axis in enumerate(AXES_TO_FIT)}
+
+    for axis in AXES_TO_FIT:
+        print()
+        effect_units = UNITS_DICT[axis]
+        effect_data = data[axis]
+
+        data_plot, resid_plot = plot_dict[axis]
+        data_plot.set(ylabel=f"{axis} ({effect_units})")
+        resid_plot.set(
+            xlabel=f"{cause} ({cause_units})", ylabel=f"{axis} fractional residuals)"
+        )
+        data_plot.set_title(f"{axis} vs. {cause}")
+
+        mean_abs_corr = np.mean(np.abs(effect_data))
+        print(f"residuals divisor = {mean_abs_corr:0.3g} (mean(abs({axis} offset))")
+
+        for ncoeffs in ncoeffs_arr:
+            coeffs, covariance = fit_one(
+                data=data, cause=cause, axis=axis, model=model, ncoeffs=ncoeffs
+            )
+            dense_modeled_data = model(dense_cause_data, *coeffs)
+            data_plot.plot(dense_cause_data, dense_modeled_data)
+            residuals = (model(cause_data, *coeffs) - effect_data) / mean_abs_corr
```

Comment:
The absolute residuals (not relative) would be the better thing to look at. They should be compared to the non-repeating errors for each axis (as defined by the repeatability requirements).